catalog/transformer/catalog-transformer-csv-common/src/main/java/ddf/catalog/transformer/csv/common/CsvTransformer.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.commons.csv.CSVPrinter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Common Library used to transform {@link Metacard}s into CSV text.
+ *
+ * @see MetacardTransformer
+ * @see Metacard
+ * @see QueryResponseTransformer
+ * @see QueryResponse
+ * @see Attribute
+ */
+public class CsvTransformer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CsvTransformer.class);
+
+  private static final MimeType CSV_MIME_TYPE;
+
+  private static final Predicate<AttributeDescriptor> attributeNotBinary =
+      attrDesc ->
+          !AttributeType.AttributeFormat.BINARY.equals(attrDesc.getType().getAttributeFormat());
+
+  private static final Predicate<AttributeDescriptor> attributeNotObject =
+      attrDesc ->
+          !AttributeType.AttributeFormat.OBJECT.equals(attrDesc.getType().getAttributeFormat());
+
+  static {
+    try {
+      CSV_MIME_TYPE = new MimeType("text/csv");
+    } catch (MimeTypeParseException e) {
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  public static BinaryContent createResponse(final Appendable csv) {
+    InputStream inputStream =
+        new ByteArrayInputStream(csv.toString().getBytes(StandardCharsets.UTF_8));
+    return new BinaryContentImpl(inputStream, CSV_MIME_TYPE);
+  }
+
+  public static Appendable writeMetacardsToCsv(
+      final List<Metacard> metacards,
+      final List<AttributeDescriptor> orderedAttributeDescriptors,
+      final Map<String, String> aliasMap)
+      throws CatalogTransformerException {
+    StringBuilder stringBuilder = new StringBuilder();
+
+    try {
+      CSVPrinter csvPrinter = new CSVPrinter(stringBuilder, CSVFormat.RFC4180);
+      printColumnHeaders(csvPrinter, orderedAttributeDescriptors, aliasMap);
+
+      metacards.forEach(
+          metacard -> printMetacard(csvPrinter, metacard, orderedAttributeDescriptors));
+
+      return csvPrinter.getOut();
+    } catch (IOException ioe) {
+      throw new CatalogTransformerException(ioe);
+    }
+  }
+
+  private static void printMetacard(
+      final CSVPrinter csvPrinter,
+      final Metacard metacard,
+      final List<AttributeDescriptor> orderedAttributeDescriptors) {
+    Iterator<Serializable> metacardIterator =
+        new MetacardIterator(metacard, orderedAttributeDescriptors);
+    printData(csvPrinter, metacardIterator);
+  }
+
+  private static void printColumnHeaders(
+      final CSVPrinter csvPrinter,
+      final List<AttributeDescriptor> orderedAttributeDescriptors,
+      final Map<String, String> aliasMap) {
+    Iterator<String> columnHeaderIterator =
+        new ColumnHeaderIterator(orderedAttributeDescriptors, aliasMap);
+    printData(csvPrinter, columnHeaderIterator);
+  }
+
+  private static void printData(final CSVPrinter csvPrinter, final Iterator iterator) {
+    try {
+      csvPrinter.printRecord(() -> iterator);
+    } catch (IOException ioe) {
+      LOGGER.debug("Failed to print the CSV data.", ioe);
```

Comment:
Would be nice to know which metacard failed - helps track down data-specific issues when it occurs in the field.

Reply:
Added a new method for this